packages/http-client-python/generator/pygen/codegen/templates/model_dpg.py.jinja2

```diff
@@ -50,7 +50,7 @@
     {% endif %}
 
 
-    {% if serializer.need_init(model) %}
+    {% if serializer.init_overloads(model) %}
```

Comment:
Hi @iscai-msft, `need_init` is needed since some models doesn't have additional properties (like [here](https://github.com/Azure/autorest.python/blob/0b55911f1c190e01990ea08b5b73cc81c2b50f94/packages/typespec-python/test/azure/generated/azure-resource-manager-resources/azure/resourcemanager/resources/models/_models.py#L127-L143) and there is will be pylint error `useless-parent-delegation` if we always add `__init__`

Reply:
To fix this issue, I think you just need to remove `(not model.internal)` from the following code:
https://github.com/microsoft/typespec/blob/a4ad7ee171866027fd90d09b80fc771780fd8ba3/packages/http-client-python/generator/pygen/codegen/serializers/model_serializer.py#L110-L111